py_zipkin/thrift/__init__.py

```diff
@@ -1,26 +1,45 @@
 import os
 import socket
 import struct
+from typing import Dict
+from typing import List
+from typing import Mapping
+from typing import Optional
+from typing import TYPE_CHECKING
 
 import thriftpy2
 from thriftpy2.protocol import TBinaryProtocol
 from thriftpy2.protocol.binary import write_list_begin
 from thriftpy2.thrift import TType
 from thriftpy2.transport import TMemoryBuffer
+from typing_extensions import TypedDict
 
 from py_zipkin.util import unsigned_hex_to_signed_int
 
 
 thrift_filepath = os.path.join(os.path.dirname(__file__), "zipkinCore.thrift")
-zipkin_core = thriftpy2.load(thrift_filepath, module_name="zipkinCore_thrift")
+
+# this is an extremely weird pattern, but since zipkinCore isn't a "real"
+# module, but the .pyi file pretends it is, we can't import it for real but we
+# can during type checking. Hence, we end up with this weird pattern.
+if TYPE_CHECKING:  # pragma: no cover
+    from . import zipkinCore
+else:
+    # load this as `zipkinCore` so that thrift-pyi generation matches
+    zipkinCore = thriftpy2.load(thrift_filepath, module_name="zipkinCore_thrift")
```

Comment:
Should I file a ticket for circling back to think about a better way to handle this situation? I think we could write a slightly cleaner pattern:
```python
pkg/protocol/pkg.thrift
pkg/protocol/__init__.py
pkg/protocol/pkg.pyi
```
...where `__init__.py` abstracts away the `TYPE_CHECKING` check and returns either a stub module or a runtime thrift module. sort of a combination of the [pattern followed in `monk`](https://sourcegraph.yelpcorp.com/python-packages/monk/-/blob/monk/util.py#L40-52) and the pattern shown here: https://github.com/unmade/thrift-pyi#user-content-quickstart

Reply:
Yeah, that's a good idea.

Reply:
CORESERV-11991